Project2015To2017/ProjectConverter.cs

```diff
@@ -148,8 +154,18 @@ private Project ProcessFile(FileInfo file, Solution solution, ProjectReference r
 				transform.Transform(project);
 			}
 
-			foreach (var transform in TransformationsToApply(project.IsModernProject))
+			foreach (var transform in TransformationsToApply())
 			{
+				if (project.IsModernProject && transform is ILegacyOnlyProjectTransformation)
```

Comment:
That's simple actually. @mungojam mentioned the need to run new/modified transformations on already converted projects. If we move this logic to transformations (I've done that at first and then realized all that) there will be no way to force execution of specific transform (except to modify the code/add new parameters to Transform interface method). Now the only thing that should be done is add necessary _force_ logic to ProjectConverter.

Reply:
If you want to rerun then you should make sure it is deterministic, right? 

I don’t see how you ever want to force anything, we should strive for deterministic transforms. If a file no longer exists (like nuspec) just no-op, that’s fine.

Reply:
Yeah, but deterministic is not something guaranteed to be preserved between versions (behavior of FileTransformation changed greatly in my earlier PR). User wants to take advantage of improved conversion behavior, so they pass some CLI argument and force execution despite the fact that it is already converted (`IsModernProject` is `true`).

Reply:
I can see all that, but I’m missing the point as to why this check can’t be done inside the transformation itself. I see what you’re trying to say about a force option, but I don’t see yet why you would need such a thing. 

I think we should assume that the 99% of users will use it as a global tool and just run it without any parameters except for a project/solution. I don’t see a lot of users sifting through the docs in order to figure out how they can do incremental runs. The 2015 -> 2017 change was a big one, it will probably be a very long time before a similar change (needing tooling) will occur.